### PR TITLE
Add use cases and manual testing instructions

### LINKED ISSUE #471: [DG] Refine manual testing instructions
Could consider omitting how to test features (since there's already the UG), or refining further
image

More info: https://nus-cs2103-ay2122s1.github.io/website/admin/tp-deliverables.html#deliverable-developer-guide-dg

### LINKED ISSUE #508: [DG] More useful content for new Developers
- [x] More diagrams (Object, Class, Sequence, Activity)
- [x] User stories
- [x] Glossary
- [x] Effort ❗

- [x] Check if all use cases are included
- [x] Manul Testing

image

### PR DESCRIPTION
**Pull request by AllardQuek**
Closes #508
Resolves #471

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -15,34 +15,37 @@ title: Developer Guide
...
 14. [Use Cases](#Use Cases)
     1. [Use Case: UC1 Purge/Delete all sample student records](#Use Case1)
     2. [Use Case: UC2 Create a student record](#Use Case2)
-    3. [Use Case: UC3 View a student record](#Use Case3)
-    4. [Use Case: UC4 Delete a student record](#Use Case4)
-    5. [Use Case: UC5 Update a student record](#Use Case5)
-    6. [Use Case: UC6 Download student records](#Use Case6)
-    7. [Use Case: UC7 Create a lab record](#Use Case7)
-    8. [Use Case: UC8 Edit a lab record](#Use Case8)
+    3. [Use Case: UC3 Create a student record](#Use Case3)
```
Comment (reddevil1313) [inline docs/DeveloperGuide.md:32]: UC3 should be Upload student records.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -841,7 +843,28 @@ Precondition: CS2100 TA opens ProgrammerError for the first time
 
     Use case resumes at step 3.
 
-### <a name="Use Case3"></a> **Use case: UC3 View a student record**
+### <a name="Use Case3"></a> **Use case 3: UC3 Upload student records**
```
Comment (shermannws) [inline docs/DeveloperGuide.md:846]: I think there is inconsistent use of "Use case" and "Use case <Number>".
Might want to consider standardising.

**Review by shermannws — CHANGES_REQUESTED** [review]
Left a comment on the standardization of "Use case <NUMBER>"! Proposed to stick with "Use case: UC<NUM>" as per [textbook](https://nus-cs2103-ay2122s1.github.io/website/se-book-adapted/chapters/specifyingRequirements.html)

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (shermannws) [reply docs/DeveloperGuide.md:846]: Propose for this line to be
`### <a name="Use Case3"></a> **Use case: UC3 Upload student records**`

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (AllardQuek) [reply docs/DeveloperGuide.md]: Fixed

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (AllardQuek) [reply docs/DeveloperGuide.md]: Should be fixed now!

**Review by reddevil1313 — APPROVED** [review]
LGTM

**Review by erwinqxy — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -967,6 +999,17 @@ Precondition: CS2100 TA opens ProgrammerError for the first time
 
   Use case ends.
 
+### <a name="Use Cas10"></a> **Use case: UC10 View dashboard of student records**
+
+**MSS**
+
+1. CS2100 TA requests to view the dashboard of student data
```
[the remark this answers is not in the document]
Comment (AllardQuek) [reply docs/DeveloperGuide.md]: Looks like we have not been consistent throughout all the use cases. Fixed.

**Review by erwinqxy — APPROVED** [review]
LGTM

**Review by shermannws — APPROVED** [review]
LGTM

**Pull request merged**